src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,573 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Mathieu Carrière
+#
+# Copyright (C) 2021 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+import numpy as np
+import itertools
+import matplotlib
+import matplotlib.pyplot as plt
+
+from sklearn.base            import BaseEstimator, TransformerMixin
+from sklearn.cluster         import DBSCAN, AgglomerativeClustering
+from sklearn.metrics         import pairwise_distances
+from scipy.spatial.distance  import directed_hausdorff
+
+from .. import SimplexTree, CoverComplex
+
+class CoverComplexPy(BaseEstimator, TransformerMixin):
+    """
+    This is a mother class for MapperComplex, GraphInducedComplex and NerveComplex.
+
+    Attributes:
+        simplex_tree (gudhi SimplexTree): simplicial complex representing the cover complex computed after calling the fit() method.
+        node_info (dictionary): various information associated to the nodes of the cover complex.
+    """
+    def __init__(self, input_name="data", cover_name="cover", color_name="color", verbose=False):
+        """
+        Constructor for the CoverComplexPy class.
+
+        Parameters:
+        """
+        self.input_name, self.cover_name, self.color_name, self.verbose = input_name, cover_name, color_name, verbose
+
+    def get_networkx(self, get_attrs=False):
+        """
+        Turn the 1-skeleton of the cover complex computed after calling fit() method into a networkx graph.
+        This function requires networkx (https://networkx.org/documentation/stable/install.html).
+
+        Parameters:
+            get_attrs (bool): if True, the color functions will be used as attributes for the networkx graph.
+
+        Returns:
+            G (networkx graph): graph representing the 1-skeleton of the cover complex.
+        """
+        try:
+            import networkx as nx
+            st = self.simplex_tree
+            G = nx.Graph()
+            for (splx,_) in st.get_skeleton(1):	
+                if len(splx) == 1:
+                    G.add_node(splx[0])
+                if len(splx) == 2:
+                    G.add_edge(splx[0], splx[1])
+            if get_attrs:
+                attrs = {k: {"attr_name": self.node_info[k]["colors"]} for k in G.nodes()}
+                nx.set_node_attributes(G, attrs)
+            return G
+        except ImportError:
+            print("Networkx not found, nx graph not computed")
+
+    def print_to_dot(self, epsv=.2, epss=.4):
+        """
+        Write the cover complex in a DOT file called "{self.input_name}.dot", that can be processed with, e.g., neato.
+
+        Parameters:
+            epsv (float): scale the node colors between [epsv, 1-epsv]
+            epss (float): scale the node sizes between [epss, 1-epss]
```

Comment:
I am not familiar with dot, is there a reason to want to scale those numbers between eps and 1-eps? Are the colors on a circle with 0=1, and epsv allows to keep some gap between the colors of min and max? Are the sizes 0 and 1 just too small/big?
`eps_color`/`eps_size` seem a bit more readable.

Reply:
Basically `epsv` and `epss` (which I replaced with `epsilon_color` and `epsilon_size`) control the variances in size and color of the nodes. Not sure if this is really useful for colors though, I created for sizes since sometimes people like to have Mapper nodes with different sizes yet not too different.